src/main/java/com/mongodb/kafka/connect/sink/writemodel/strategy/ReplaceOneDefaultStrategy.java

```diff
@@ -44,6 +45,12 @@
                     new DataException(
                         "Could not build the WriteModel,the value document was missing unexpectedly"));
 
-    return new ReplaceOneModel<>(new BsonDocument(ID_FIELD, vd.get(ID_FIELD)), vd, REPLACE_OPTIONS);
+    BsonValue idValue = vd.get(ID_FIELD);
+    if (idValue == null) {
```

Comment:
When refactoring the error sceanrios I noticed that these errors would cause IllegalArgumentExceptions when creating the WriteModels - so I normalized them to DataExceptions.